src/scripts/migrate.py

```diff
@@ -0,0 +1,141 @@
+"""Slurp migration pipeline
+
+TODO's:
+  -
+"""
+import os.path
+from argparse import ArgumentParser
+from typing import Dict, List
+
+import pandas as pd
+import oaklib
+from oaklib.implementations import ProntoImplementation
+from oaklib.resource import OntologyResource
+
+
+# TODO: implement this func:
+# todo: IDs should be int or str? prolly str
+def determine_next_available_mondo_id(min_id: str, mondo_termlist_df: pd.DataFrame) -> str:
+    """Starting from `min_id`, count up and check until finding the next ID."""
+    next_id = str(0)
+    return next_id
+
+
+def _get_direct_parents(curie: str, ontology: ProntoImplementation):
+    """Get CURIEs of parents of a class
+    todo: `ontology` typing: should probably be typed agnostic of the implementation. But it inherits from so
+     much stuff, I'm not sure which if any of these to use. I was expecting `BasicOntologyInterface`, but it doesn't
+     inherit from that: RelationGraphInterface, OboGraphInterface, ValidatorInterface, SearchInterface,
+     SubsetterInterface, MappingProviderInterface, PatcherInterface, SemanticSimilarityInterface, MetadataInterface,
+     DifferInterface, ABC
+    """
+    # TODO: How to get just the *direct* parents? (a) func for that? (b) look at relationships and see if subClass?
+    # rels = ontology.get_outgoing_relationships_by_curie(curie)  # works for ProntoImplementation
+    rels = ontology.relationships([curie])
+    p = []
+    for rel, parents in rels.items():
+        # print(f'  {rel} ! {ontology.get_label_by_curie(rel)}')
+        for parent in parents:
+            # print(f'    {parent} ! {ontology.get_label_by_curie(parent)}')
+            p.append(parent)
+    return p
+
+
+def _get_all_term_curies(ontology: ProntoImplementation, is_omim=False) -> List[str]:
+    """Get all terms as CURIEs
+    todo: Will we need prefix_map from our config `.yml`s? Or can OAK handle automatically?
```

Comment:
@matentzn Do we need to add our `metadata/%.yml` files as a parameter to this script, in order to get `base_prefix_map`? That way we can get the list of terms that are 'owned' by that ontology? I think we do.

Reply:
Why dont you simply pass in `mirror-signature-mondo` which has all the terms?

Reply:
@matentzn Sorry for my super-late response here. I'm not trying to get Mondo terms; I'm trying to get the terms from ontologies we are slurping.

Is `reports/mirror_signature-mondo.tsv` supposed to have all of the terms for all ontologies we've imported? I did not think that was the case. If it is supposed to be the case, then we have a problem. I just checked, and I don't see anything for `omim` or `ordo`, `icd10cm`, or most of the ontologies we are slurping. I do see some NCIT terms in there, though.

In addition, if the goal of slurping that we are supposed to be importing new terms that _aren't yet in Mondo_, then it doesn't make sense to be querying this file for those terms.